server/supernova/audio_backend/portaudio_backend.hpp

```diff
@@ -84,35 +85,14 @@ class portaudio_backend : public detail::audio_backend_base<sample_type, float,
             report_error(device_number);
 
         printf("Available Audio Devices:\n");
-        for (int i = 0; i != device_number; ++i) {
-            const PaDeviceInfo* device_info = Pa_GetDeviceInfo(i);
-            if (device_info) {
-                printf("%d: %s (%d inputs, %d outputs)\n", i, device_info->name, device_info->maxInputChannels,
-                       device_info->maxOutputChannels);
-            }
+        for (int i = 0; i < device_number; i++) {
+            const PaDeviceInfo* pdi = Pa_GetDeviceInfo(i);
+            printf("%d: %s  (%d ins, %d outs)\n", i, GetPaDeviceName(i).c_str(), pdi->maxInputChannels,
```

Comment:
perhaps you want to indent these lines by 2 spaces to match the other output? also, please remove the extra space in `%s  (%d ins`

Reply:
Perhaps? I'll write more about posting below, but quickly this is the status of posting now:
- scsynth + coreaudio:
`   ID : "name"` (3-space indent)
- scsynth + portaudio:
`  - name   (device #ID with X ins Y outs)` (2-space indent, no quotes, device ID in parentheses; also two spaces before parenthesis)
- supernova + portaudio in this PR:
`0: Built-in Microphone (2 ins, 0 outs)` (no indent, device ID at the beginning, I removed double space already)